### DISCUSSION TITLE
Is there a way to run this for with Firefox Portable?

### DISCUSSION THREAD

**Discussion by dajix350** [filips123/PWAsForFirefox#78] [Q&A] [answered]
I am making an ultimate USB for my apps, because I am constantly switching between devices nowadays. I was wondering if there is a way to run this in a portable environment with PortableApps.com's Firefox portable.

**Comment by filips123** [comment]
It could probably work in theory, but I'm not sure what the best way to implement this is. There are currently two problems:

* Paths to Firefox installation, profiles, and config are currently set inside the user profile and cannot be changed without recompiling the program, so they cannot be moved to USB. In order to support portable installations, we would need some way to specify all paths at run-time and also make sure they work with removable devices.

* Some features of system integrations (for example, taskbar icons, shortcut actions, launching from start menu...) probably won't work if PWAs are not registered/installed to the OS.

There are currently two possible workarounds:

* Set up Firefox Account/Sync, so data within PWAs will be synced between all devices. However, you will still need to manually install each PWA into each device to set it up, and data that Firefox Account cannot sync won't be synced.

* Set up symlinks between [standard FirefoxPWA directories](https://github.com/filips123/FirefoxPWA/blob/main/native/src/directories.rs#L23-L68) and some location on your USB. However, you will still need to manually set it up on each device, and it's kinda hacky, so it may not always work correctly.

**Comment by filips123** [comment, marked as answer]
I forgot to mention it here when support for this was added (one year ago...), but you can PWAsForFirefox also provides `.paf.exe` packages for PortableApps.com.

You can download it from the releases page and install it using the PortableApps.com Platform. To connect it with the extension, you have to run the PWAsForFirefox background program. Installed web apps will then also be added to the PortableApps.com menu. More documentation is provided in the `help.html` file of the package.
